### ISSUE TITLE
machines questions: Virtual machines and oVirt machines

### ISSUE THREAD

**Issue by kenzhaoyihui** [cockpit-project/cockpit#8101]
Just three problem:

1. cockpit-ovirt package is the oVirt machines plugin which display in the cockpit, cockpit-machine is the virtual machines plugin which display in the cockpit. From my point of view, the ovirt machines manage the VM which created by RHVM, the virtual machines manage the VM which created by libvirt, So, do both them can display on the cockpit?

2. Could the ovirt machines manage the VM which created by libvirt, or the virtual machines manage the VM which created by RHVM?

3. Could cockpit provide the UI(cockpit-machine) to create the VM by libvirt api?

Thanks!
Yihui

**Comment by mareklibra** [comment]
TL;DR : use 
- cockpit-ovirt on an oVirt host (RHV-H)
- cockpit-machines on pure libvirt machine

There's no packaging conflict when installing both of them but the cockpit-ovirt has higher priority at run-time. Both plugins are mapped to the same context root `/machines`.

> cockpit-ovirt package is the oVirt machines plugin which display in the cockpit, cockpit-machine is the virtual machines plugin which display in the cockpit. From my point of view, the ovirt machines manage the VM which created by RHVM, the virtual machines manage the VM which created by libvirt, So, do both them can display on the cockpit?

These two plugins intentionally can't be running at the same time.
When particular machine is a RHV-H (member of oVirt/RHV cluster), the cockpit-ovirt is better fit here.
Any running VM is imported to oVirt as 'external' one. Access to Libvirt is secured on RHV-H to discourage anyone to interact with libvirtd.
It's low priority, but there might be possibility to manage purely Libvirt VMs via cockpit-ovirt in the future. But recently this makes probably little sense.

> Could the ovirt machines manage the VM which created by libvirt, or the virtual machines manage the VM which created by RHVM?

The cockpit-machines (Libvirt) is base package for cockpit-ovirt.
The user can run cockpit-machines on RHV-H but write operations are blocked (see above). Conceptually, it would make no sense to manage oVirt-originated machines via Libvirt directly unless troubleshooting. 

To run cockpit-ovirt within pure libvirt environment is possible. Write operations are not yet implemented (see above), but might be in the future if real use case is found. 
Anyway, the notion of "cluster" within cockpit-ovirt running on a non-oVirt host might be confusing from the user perspective. The plugin was not intended to be deployed this way.

> Could cockpit provide the UI(cockpit-machine) to create the VM by libvirt api?

Yes, wip: https://github.com/cockpit-project/cockpit/pull/7820

**Comment by kenzhaoyihui** [comment]
Clearly now. Thanks!
